core/src/test/scala/ai/lum/odinson/foundations/TestQueryCompiler.scala

```diff
@@ -27,54 +46,182 @@ class TestQueryCompiler extends BaseSpec {
+      QCHelper.wrapQuery(
+        QCHelper.spanTermQuery(
+          QCHelper.termFoo
+        )
+      )
+    )
+    qc.mkQuery("(?=foo)") shouldEqual (result) // test negative lookahead
+    // test negative lookahead
+    val result1 = new OdinNotQuery(QCHelper.allNGRams0, result, "norm")
+    qc.mkQuery("(?!foo)") shouldEqual (result1)
   }
 
   it should "compile concatenation and disjunctives correctly" in {
     // get fixture
     val ee = getExtractorEngine
     val qc = ee.compiler
     // test or
-    qc.mkQuery("foo|bar")
-      .toString shouldEqual ("OrQuery([Wrapped(norm:foo),Wrapped(norm:bar)])")
+    qc.mkQuery("foo|bar") shouldEqual (new OdinOrQuery(
+      List(
+        QCHelper.wrapedFooQuery,
+        QCHelper.wrapedBarQuery
+      ),
+      "norm"
+    ))
     // triple or
-    qc.mkQuery("a|b|c")
-      .toString shouldEqual ("OrQuery([Wrapped(norm:a),Wrapped(norm:b),Wrapped(norm:c)])")
+    qc.mkQuery("foo|bar|foobar") shouldEqual (new OdinOrQuery(
+      List(
+        QCHelper.wrapedFooQuery,
+        QCHelper.wrapedBarQuery,
+        QCHelper.wrapedFoobarQuery
+      ),
+      "norm"
+    ))
+    // test or with equal strings (should not return a OrQuery)
+    qc.mkQuery("foo|foo") shouldEqual (QCHelper.wrapedFooQuery)
+    // test or with equal strings (should ignore the repeated element)
+    qc.mkQuery("foo|foo|bar") shouldEqual (new OdinOrQuery(
+      List(
+        QCHelper.wrapedFooQuery,
+        QCHelper.wrapedBarQuery
+      ),
+      "norm"
+    ))
     // test concatenation
-    qc.mkQuery("(a)(b)")
-      .toString shouldEqual ("Concat([Wrapped(norm:a),Wrapped(norm:b)])")
+    qc.mkQuery("(foo)(bar)") shouldEqual (new OdinConcatQuery(
+      List(
+        QCHelper.wrapedFooQuery,
+        QCHelper.wrapedBarQuery
+      ),
+      "norm",
+      "numWords"
+    ))
     // test triple concatenation
-    qc.mkQuery("(a)(b)(c)")
-      .toString shouldEqual ("Concat([Wrapped(norm:a),Wrapped(norm:b),Wrapped(norm:c)])")
+    qc.mkQuery("(foo)(bar)(foobar)") shouldEqual (new OdinConcatQuery(
+      List(
+        QCHelper.wrapedFooQuery,
+        QCHelper.wrapedBarQuery,
+        QCHelper.wrapedFoobarQuery
+      ),
+      "norm",
+      "numWords"
+    ))
+    // test repeated double concatenation
+    qc.mkQuery("(foo)(foo)") shouldEqual (new OdinConcatQuery(
+      List(
+        QCHelper.wrapedFooQuery,
+        QCHelper.wrapedFooQuery
+      ),
+      "norm",
+      "numWords"
+    ))
   }
 
   it should "compile graph traversals correctly" in {
     // get fixture
     val ee = getExtractorEngine
     val qc = ee.compiler
-    // outgoing
+    // testing .toString
+    val pattern = """
+      trigger = bar
+      object = >nsubj
+      """
+    // test toString 
+    qc.compileEventQuery(pattern).toString shouldEqual (QCHelper.eventTestQuery.toString)
+    // test toString 
+    qc.compileEventQuery(pattern) shouldEqual (QCHelper.eventTestQuery)
+    
+    // TODO why is this failing?
```

Comment:
We shouldn't try to fix this, we should replace it with the same style you're using above, i.e., making the objects and comparing them